### ISSUE TITLE
Exception: Invalid uid 

### ISSUE THREAD

**Issue by aniketor** [centricular/gstwebrtc-demos#156]
I have been trying to reload the page and stream my RTSP pipeline. But whenever I reload the page the server is disconnected and it cannot find the previous peer ID. Has anyone encountered this problem? Why doesn't it restart?

**Comment by thaytan** [comment]
Each time you reload the page, it's a brand new connection to the server, and the javascript code in the browser assigns you a new random peer id. 

If you want something more persistent, you need to host the HTML/JS code yourself and modify the webrtc.js script to use a fixed peer ID.

**Comment by aniketor** [comment]
Already using a fix peer ID in the js. However, after a refresh when it disconnects and reconnects to the signalling server again this fix ID of js is seen invalid. I'm aware there is a clean up of peer IDs in server after disconnect, but why the invalid ID when there is a new connection established.

**Comment by thaytan** [comment]
@nirbheek might be able to comment better on what's happening on the server side.

**Comment by nirbheek** [comment]
What changes did you make to `webrtc.js` to make it use a fixed peer id? You should only need to set the `default_peer_id` variable.

AFAICT the most likely way that invalid ID error can happen on reconnect is if the previous connection wasn't closed properly before a new one was attempted. Does the error still happen if you wait for a little bit and retry?

**Comment by aniketor** [comment]
Hi @nirbheek 
On the webrtc.js side I have set only `default_peer_id` . In the same file I have disabled getmedia as don't want to receive the stream, I am trying only send. I am running the server and webrtc-sendrecv on two threads parallely. I added a sleep so that the clean up happens and connection closes properly. Now it works after refresh but only for a couple of times. During 3rd time it opens up the session from one peer to the fixed peeer, but doesn't send any ice candidates, it just pauses.

log:
Connection to peer ('127.0.0.1', 54560) closed, exiting handler
Removed all peer IDs-------------------
Disconnected from peer '1111' at ('127.0.0.1', 54554)
Removed all peer IDs-------------------
Connected to ('127.0.0.1', 54564)
Registered peer '1111' at ('127.0.0.1', 54564)
Connected to ('127.0.0.1', 54566)
Registered peer '1234' at ('127.0.0.1', 54566)
'1234' command 'SESSION 1111'
Session from '1234' (('127.0.0.1', 54566)) to '1111' (('127.0.0.1', 54564))
